dashboard/app/controllers/pd/workshop_daily_survey_controller.rb

```diff
@@ -225,6 +225,67 @@ def new_post
       render :new_general
     end
 
+    # GET /pd/workshop_survey/csf/:survey_name/:enrollment_code
+    def new_csf
+      params[:survey_name] = params[:survey_name]&.downcase
+      return render_404 unless CSF_SURVEY_INDEXES.key? params[:survey_name]
+
+      if params[:survey_name] == PRE_DEEPDIVE_SURVEY
+        # TODO: what if we don't require log-in? what if pre-ws survey can be annonymous
+        workshop = Workshop.
```

Comment:
Will do

Reply:
Actually on a closer look, we have many things in this file that are in Pd module such as `WorkshopDailySurvey`, `WorkshopFacilitatorDailySurvey` and all the constants. If we want to be consistent we will have to append `Pd::` to all of them. However, it doesn't add any value beside of looking like other files outside of Pd module. I think it might even add a slight confusion when an item inside a namespace is prefix-ed with the namespace itself.

Reply:
Until recently, we had two versions of the `Workshop` class, one inside `Pd` and one outside, so our convention has been to always specify the Pd to disambiguate inline without having to check whether the file is in Pd. Now that we've removed the old version of workshops, we probably don't need to be as rigid - this just looks weird to me because for so long we had to be careful about which we use, and every other reference throughout our codebase (including within Pd files) uses the prefix. Personally, I still like that convention, but if we don't feel like it is serving us anymore we can change it.